sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -186,5 +186,7 @@ protected internal override object InstrumentOperation(Type operationType, objec
                 new GetOriginalInterceptor(operation),
                 new OperationInterceptor(Mode == RecordedTestMode.Playback));
         }
+
+        protected TestRetryHelper TestRetryHelper => new TestRetryHelper(Mode == RecordedTestMode.Playback);
```

Comment:
What if `Mode` changes? Should this pass down the `Mode` reference instead?

Reply:
Would it ever change in the middle of retrying?

Reply:
I may be wrong, but when we try to re-record, don't we change the mode of the current instance? So this would be constructed once in Playback mode, then used again after switching to Record mode.

Reply:
The property returns a new instance of TestRetryHelper  every time.

Reply:
Ah - that part I missed. 👍